gui/lib/widgets/link.dart

```diff
@@ -37,3 +47,41 @@ LaunchableUri _toLaunchable<T>(T uri) {
   if (uri is Uri) return LaunchableUri(uri);
   throw ArgumentError('Unsupported type $T: $uri');
 }
+
+/// A clickable link widget with a trailing icon.
+///
+/// Extends [Link] to display a theme-aware icon alongside the link text.
+/// The icon is positioned to the right of the text with [AppSpacing.spacing2] spacing.
+///
+/// Accepts either a [DynamicThemeImage] object via [icon] or a string path
+/// via [iconPath]. Exactly one must be provided. Use [Link] directly if no
+/// icon is needed.
+final class IconLink<T> extends Link<T> {
+  final DynamicThemeImage _icon;
+
+  IconLink({
+    super.key,
+    required super.title,
+    required super.uri,
+    super.size,
+    DynamicThemeImage? icon,
+    String? iconPath,
+  }) : assert(
+         (icon != null) ^ (iconPath != null),
+         'Exactly one of icon or iconPath must be provided',
```

Comment:
Is `icon` approach used somewhere? 🤔 I see only `iconPath` used

Reply:
Aah, ok. Then maybe we could remove it in both places, because it's a dead code now :man_shrugging: